src/identity_policy_provider.py

```diff
@@ -0,0 +1,208 @@
+                                        }
+                                    }
+                                }, {
+                                    "required": ["AWS"],
+                                    "properties": {
+                                        "AWS": {
+                                            "type": "array",
+                                            "items": {
+                                                "type": "string"
+                                            }
+                                        }
+                                    }
+                                }]
+                            },
+                            "Action": {
+                                "type": "array",
+                                "items": {
+                                    "type": "string",
+                                    "enum": ["ses:SendEmail", "ses:SendRawEmail"]
+                                }
+                            }
+                        }
+                    }
+                },
+                "Version": {
+                    "type": "string",
+                    "description": "of the policy document"
+                }
+            }
+        }
+    }
+}
+
+
+class IdentityPolicyProvider(ResourceProvider):
+
+    def __init__(self):
+        super(IdentityPolicyProvider, self).__init__()
+        self.request_schema = request_schema
+        self.ses = boto3.client('ses')
+
+    def convert_property_types(self):
+        self.heuristic_convert_property_types(self.properties)
+
+    def create(self):
+        existing_policy = self.get_policy(self.get('Identity'), self.get('PolicyName'))
+        if existing_policy is not None:
+            self.fail(f"identity policy {self.get('PolicyName')} already exists")
+            return
+
+        desired_policy_document = PolicyDocument.from_dict(self.get('PolicyDocument'))
+        self.put_policy(desired_policy_document)
+
+    def update(self):
+        if self.physical_resource_id != self.get('PolicyName'):
+            self.fail("changing property PolicyName is not allowed")
+            return
+
+        current_policy = self.get_policy(self.get('Identity'), self.get('PolicyName'))
+        if current_policy is None:
+            self.fail(f"identity policy {self.get('PolicyName')} does not exist")
+            return
+
+        current_policy_document = PolicyDocument.from_json(current_policy)
+        desired_policy_document = PolicyDocument.from_dict(self.get('PolicyDocument'))
+        if current_policy_document != desired_policy_document:
+            self.put_policy(desired_policy_document)
+
+    def put_policy(self, policy_document):
+        try:
+            self.ses.put_identity_policy(
+                Identity=self.get('Identity'),
+                PolicyName=self.get('PolicyName'),
+                Policy=policy_document.to_json())
+            self.physical_resource_id = self.get('PolicyName')
```

Comment:
To support renaming the policy, use a physical_resource_id which consists of the Identity and the Policy Name. This will ensure that you will get a delete if either of these keys change.

```python
self.physical_resource_id = f'{self.identity}/@{self.policy_name}'
```